docs/CONTRIBUTING.md

```diff
@@ -7,29 +7,134 @@ sure you follow the steps below:
-use the `viv` executable in the repository.
+    sudo pacman -S --needed yarn make zip
+
+**Fedora and derivatives:**
+
+    sudo dnf install yarn make zip
+
+> [!NOTE]
+> The version of **Node.js** shipped with some Linux distributions will fail to
+> build **Vivify**, see the section below to install the latest node from **nvm**
+
+## Installing Node.js
+
+Node.js may be available with your OS or in your distros package manager.
+Alternatively you can install the latest version of Node.js using **[Node
+Version Manager](https://github.com/nvm-sh/nvm)** (**nvm**)
+
+To install **nvm** use:-
+
+    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+
+You will then need to reload your terminal by either closing it and opening a
+new one or by re-sourcing your run commands (bash eg: `source ~/.bashrc`)
+
+Now you can install the latest version of Node.js:-
+
+    nvm install node
+
+> [!TIP]
+> You can return to the system version of node using `nvm use system`
+>
+> For more details on nvm usage or for troubleshooting installing nvm, visit the
+> nvm [Documentation][1] page
+>
+> [1]:<https://github.com/nvm-sh/nvm?tab=readme-ov-file#node-version-manager--->
+
+## Building Vivify
+
+First clone and open the **Vivify** repository
+
+    git clone https://github.com/jannis-baum/Vivify.git
+    cd Vivify
+
+Run `yarn` to download all node dependencies
+
+    yarn
+
+Then build **Vivify** by running `make <OS-family>` where *OS-family* is your
+operating system (Linux or MacOS)
+
+**MacOS:**
+
+    make macos
+
+**Linux:**
+
+    make linux
+
+If you want to install **Vivify** for use outside of your development
+environmentm you can use `make install` but first you need to run the
+`configure <install-dir>` script, eg:-
+
+    ./configure ~/.local/bin
+    make install
+
+Once installed you can launch **viv** by calling `viv <some-path-or-file>`
+
+> [!TIP]  
+> Ideally the install location should be included in your $PATH variable
+
+## Running Vivify for Development
+
+Once you have built **Vivify** or installed it using the pre-built binaries, you
+can then run the **Vivify** server in development mode to try it out with
+live-updates as you are coding.
+
+Using `yarn dev` to run **Vivify** in dev mode will:-
+
+1. Run `vivify-server` on port `3000` instead of the usual port of `31622` so
+   they can be run side by side;
+2. Automatically reload when you make changes to the code; and
+3. Unlike the installed version, not shut down when there are no connected
+   clients.
+
+First, use `yarn` to make sure all node dependencies are up to date
+
+    yarn
+
+Then start the **Vivify** server in dev mode
+
+    yarn dev
+
+Once the development server is running, you can connect as many instances as you
+like using:-
+
+    yarn viv .
+
+> [!TIP]
+> Using `yarn viv` will connect to the development server on port 3000 instead
+> of running the installed server, and will use the viv executable in the
+> repository
```

Comment:
This changed recently: `yarn viv` doesn't use the `viv` executable at all anymore, instead it just calls the server that now handles (almost) all CLI and options. This initially changed in https://github.com/jannis-baum/Vivify/pull/90 and further updates were made in https://github.com/jannis-baum/Vivify/pull/112 in case you want details

Reply:
I noticed something changed, but I didn't understand what was going on. I thought I must have broke something. I'll go look these up for context.

The other comments you raise are just me being stupid too late at night, I'll fix them up.

Reply:
Yes, no worries at all! The architecture of Vivify is probably a bit complicated to understand by just looking at the code because of the lazy starting and terminating of the server and all that. That's why I've been wanting to write a doc for that as well in #119.

Reply:
Let me know if I can help with that, obviously first I'll also need to understand it better.

Even if you throw up bullet points type rough draft, I can help flesh it out and prettify it. :)

Reply:
Nice! This is actually probably a great way of doing it because if I write it entirely myself it might end up not being detailed enough to understand for new people. Will definitely get back to you on this one :)